src/git/s.ml

```diff
@@ -106,33 +106,17 @@ module type DEFLATE = sig
   val eval       : src:Cstruct.t -> dst:Cstruct.t -> t -> [ `Flush of t | `Await of t | `Error of (t * error) | `End of t ]
 end
 
-module type IDIGEST = sig
-  type t
-  type ctx
-
-  val init   : unit -> ctx
-  val feed_c : ctx -> Cstruct.t -> ctx
-  val feed_s : ctx -> string -> ctx
-  val feed_b : ctx -> bytes -> ctx
-  val get    : ctx -> t
-
-  val length : int
-end
-
 module type HASH = sig
-  include BASE with type t = private string
-
-  module Digest : IDIGEST with type t = t
-
-  val get : t -> int -> char
+  include Digestif.S
 
-  val to_string : t -> string
-  val of_string : string -> t
-
-  type hex = string
+  val feed_cstruct: ctx -> Cstruct.t -> ctx
+  val compare: t -> t -> int
+  val hash: t -> int
+  val equal: t -> t -> bool
```

Comment:
We need to choose between `eq` and `equal` but have both is not a good idea. I think `equal` should be the best and I can change API of `Digestif.S` to provide `equal` instead `eq` but we need to choose one - then if we choose `equal`, we need to update this PR to replace any `H.eq` to `H.equal`.